mtags/src/main/scala/scala/meta/internal/metals/CompilerOffsetParams.scala

```diff
@@ -37,3 +38,42 @@ object CompilerOffsetParams {
     )
   }
 }
+
+case class CompilerRangeParams(
+    uri: URI,
+    text: String,
+    offset: Int,
+    endOffset: Int,
+    token: CancelToken = EmptyCancelToken
+) extends RangeParams {
+  def toCompilerOffsetParams: CompilerOffsetParams =
+    CompilerOffsetParams(
+      uri,
+      text,
+      offset,
+      token
+    )
+}
+
+object CompilerRangeParams {
+
+  def fromPos(pos: Position, token: CancelToken): CompilerRangeParams = {
+    val syntax = pos.input.syntax
+    val uri =
```

Comment:
Could you move this to a method? It's also repeated above.

Reply:
sure